### PR TITLE
[MM-58317]: disable plugin if skip market place if prepackage flg is set

### PR DESCRIPTION
**Pull request by Rajat-Dabade**
Fixes: https://mattermost.atlassian.net/browse/MM-58317

This PR adds support for a flag named `skip_marketplace_if_prepackaged`, if set then the prepackaged plugin will not be shown up in the marketplace plugin. 

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: server/Makefile [hunk 1]
```diff
@@ -153,6 +153,7 @@ PLUGIN_PACKAGES += mattermost-plugin-nps-v1.3.3
 PLUGIN_PACKAGES += mattermost-plugin-servicenow-v2.3.4
 PLUGIN_PACKAGES += mattermost-plugin-zoom-v1.8.0
 PLUGIN_PACKAGES += mattermost-plugin-ai-v0.6.3
+PLUGIN_PACKAGES += mattermost-plugin-focalboard-v8.0.0
```
Comment (marianunez) [inline server/Makefile:156]: Let's split this change out to a separate PR that will prepackage boards given that is not related directly to the rest of the changes here.

File: server/public/model/manifest.go [hunk 2]
```diff
@@ -188,6 +188,9 @@ type Manifest struct {
 
 	// Plugins can store any kind of data in Props to allow other plugins to use it.
 	Props map[string]any `json:"props,omitempty" yaml:"props,omitempty"`
+
+	// To show the plugin in market place, if set to true then hide the plugin from market place plugin list
+	DisableInMarketPlace bool `json:"disable_in_marketplace,omitempty" yaml:"disable_in_marketplace,omitempty"`
```
Comment (marianunez) [inline server/public/model/manifest.go:193]: Let's be specific that this is to hide from the marketplace for prepackaged plugins: 

```suggestion
	// If set to true then skip listing the plugin in marketplace if it is prepackaged in the server
	SkipMarketplaceIfPrepackaged bool `json:"skip_marketplace_if_prepackaged,omitempty" yaml:"skip_marketplace_if_prepackaged,omitempty"`
```

**Review by marianunez — CHANGES_REQUESTED** [review]
Also, can we add a test case for this new setting to make sure it is returning as expected from the marketplace here: https://github.com/mattermost/mattermost/blob/master/server/channels/api4/plugin_test.go#L1171

**Review by harshilsharma63 — APPROVED** [review]
Other than the comment specified by @marianunez , looks good!

File: server/Makefile (hunk 1 above)
Comment (Rajat-Dabade) [reply server/Makefile]: Sure.

File: server/public/model/manifest.go (hunk 2 above)
Comment (Rajat-Dabade) [reply server/public/model/manifest.go]: Okay

**Comment by Rajat-Dabade** [comment]
As this was more related to `focalboard` prepackaging, and `focalboard` is already added as Transitionally prepackaged plugins (https://github.com/mattermost/mattermost/pull/24225), it will get filtered from prepackaged plugins, so we don't need to write separate logic for it. Closing this PR.
